docs/dev-environment-setup.mdx

```diff
@@ -1,3 +1,7 @@
+import { Meta } from "@storybook/addon-docs";
```

Comment:
**comment**: these changes make the contents of this mdx file appear on storybook, but I think this was a feature, rather than a bug, since it's specifically intended for new developers on the team and probably shouldn't be public-facing. I assume it was designed to be read on Github.

I accept that including this with all the storybook-related `.mdx` files in the `docs` folder is confusing so we could probably move it outside, and maybe we don't even need it any more full-stop - those discussions can come later. But unless I've missed something I don't think we should be adding it to the public docs on storybook.

Reply:
Yeah the old `.md` files in directory for targeted towards new contributors rather than Carbon consumers. Perhaps it would be better to move the `.md` files to a new separate `/contributing/` directory instead?

Reply:
**suggestion**: perhaps it would be better to move the `.md` files to a new separate `/contributing/` directory instead in a separate PR?

Reply:
I like the idea of the `/contributing/` directory, but I'm  not sure why the files still now seem to be `.mdx` rather than plain `.md` as they were before. They don't seem to show up in the Storybook docs any more, which is good, but when checking them on Github it seems to not be able to cope with the `.mdx` format (see screenshot  of what I'm seeing currently at https://github.com/Sage/carbon/blob/FE-5749-storybook-upgrade/contributing/dev-environment-setup.mdx) - a plain `.md` file would get formatted properly. 
![image](https://github.com/Sage/carbon/assets/25186376/0ff300be-36ac-445d-8d4a-7c27dd731f98)